mito-ai/src/tests/AiChat/ChatMessage.test.tsx

```diff
@@ -0,0 +1,355 @@
+            const messageElement = screen.getByText('Hello, can you help me with pandas?');
+            const messageContainer = messageElement.closest('.message');
+            expect(messageContainer).toHaveClass('message-user');
+        });
+
+        it('renders an assistant message correctly', () => {
+            renderChatMessage({
+                message: createMockMessage('assistant', 'Yes, I can help with pandas. What do you need?'),
+                messageType: 'openai message'
+            });
+
+            // Check for the assistant message text and class
+            expect(screen.getByText('Yes, I can help with pandas. What do you need?')).toBeInTheDocument();
+            const messageElement = screen.getByText('Yes, I can help with pandas. What do you need?');
+            const messageContainer = messageElement.closest('.message');
+            expect(messageContainer).toHaveClass('message-assistant-chat');
+        });
+
+        it('renders a code block in an assistant message', () => {
+            renderChatMessage({
+                message: createMockMessage('assistant', 'Here is a pandas example:\n```python\nimport pandas as pd\ndf = pd.DataFrame({"A": [1, 2, 3]})\nprint(df)\n```'),
+                messageType: 'openai message'
+            });
+
+            // Check for the text part
+            expect(screen.getByText(/Here is a pandas example:/)).toBeInTheDocument();
+
+            // The code part will be rendered in the real component
+            const container = screen.getByText(/Here is a pandas example:/).closest('.message');
+            expect(container).toBeInTheDocument();
+            expect(container).toHaveClass('message-assistant-chat');
+        });
+
+        it('renders an agent planning message correctly', () => {
+            renderChatMessage({
+                message: createMockMessage('assistant', 'I need to think about this problem...'),
+                messageType: 'openai message:agent:planning'
+            });
+
+            // Check for the planning message
+            expect(screen.getByText('I need to think about this problem...')).toBeInTheDocument();
+            const messageElement = screen.getByText('I need to think about this problem...');
+            const messageContainer = messageElement.closest('.message');
+            expect(messageContainer).toHaveClass('message-assistant-chat');
+        });
+    });
+
+    describe('Error Messages', () => {
+        it('renders an error message when there is a connection error', () => {
+            renderChatMessage({
+                message: createMockMessage('assistant', 'Connection error message'),
+                mitoAIConnectionError: true,
+                mitoAIConnectionErrorType: 'timeout'
+            });
+
+            // Error message should be visible
+            expect(screen.getByText('Connection error message')).toBeInTheDocument();
+        });
+    });
+
+    describe('Interaction', () => {
```

Comment:
Maybe there's a different name than "Interaction", not sure exactly what this category is supposed to cover